### ISSUE TITLE
PrivateAddressCheck.only_public_connections doesn't work properly after migrating from 0.4.1 to 0.5.1

### ISSUE THREAD

**Issue by bozydar** [jtdowney/private_address_check#7]
This example doesn't work anymore.
```
PrivateAddressCheck.only_public_connections do
  Net::HTTP.get_response(URI.parse("http://192.168.1.1"))
end
# => raises PrivateAddressCheck::PrivateConnectionAttemptedError
```

**Comment by bwillis** [comment]
Did you get a `Net::OpenTimeout` or `Errno::ECONNREFUSED` error? Might be addressed in #6

**Comment by bozydar** [comment]
A `Net::OpenTimeout`. Your comment about the server existence in #6 makes a lot of sense for me.
Thanks!
